### PR TITLE
@task PDA-1244 PUT /items/{itemId} - build graphQL query

### PR DESCRIPTION
**Pull request by rliu-parasoft**

### CODE REVIEW

File: src/test/java/com/parasoft/demoapp/graphql/ItemGraphQLDataFetcherTest.java [hunk 1]
```diff
@@ -108,4 +112,169 @@ public void test_getItems_incorrectAuthentication() throws IOException {
                 .and()
                 .assertThatField(ITEM_DATA_JSON_PATH).isNull();
     }
+
```
Comment (rliu-parasoft) [inline src/test/java/com/parasoft/demoapp/graphql/ItemGraphQLDataFetcherTest.java:115]: (TODO - need to research why it fails) - "test_getItems_normal"

File: src/main/resources/static/schema.graphql [hunk 2]
```diff
@@ -115,6 +117,14 @@ type Item {
     lastAccessedDate: DateTime
     categoryId: Int
 }
+input ItemDTO {
```
Comment (yshi-parasoft) [inline src/main/resources/static/schema.graphql:120]: Please change all fields here except "imagePath" to required fields to be consistent with corresponding REST API.

File: src/test/resources/graphql/items/updateItemById.graphql [hunk 3]
```diff
@@ -0,0 +1,12 @@
+mutation updateItemById($itemId: ID!, $itemDTO: ItemDTO) {
```
Comment (yshi-parasoft) [inline src/test/resources/graphql/items/updateItemById.graphql:1]: Please update the field to be required as "!ItemDTO"

File: src/test/java/com/parasoft/demoapp/graphql/ItemGraphQLDataFetcherTest.java (hunk 1 above)
Comment (yshi-parasoft) [reply src/test/java/com/parasoft/demoapp/graphql/ItemGraphQLDataFetcherTest.java:115]: @rliu-parasoft You can take a look at how Alfie reset the database for specific tests in this pull request.
https://github.com/parasoft/parasoft-demo-app/pull/119/files#diff-645b641a7060108ad2ed9c095776770a804bf16056cb20b0464be18dc4cb9472

**Comment by yshi-parasoft** [comment]
Please merge conflict and introduce Long scalar for the itemId.
After merging from main, please include empty itemId test case.

**Comment by rliu-parasoft** [comment]
This pull request is replaced with [PUT /items/{itemId} - build graphQL query](https://github.com/parasoft/parasoft-demo-app/pull/149/files)

**Pull request closed without merging**